Code/Tools/AssetProcessor/native/AssetManager/SourceAssetReference.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <native/AssetManager/SourceAssetReference.h>
+
+namespace AssetProcessor
+{
+    SourceAssetReference::SourceAssetReference(AZ::IO::PathView absolutePath)
+    {
+        IPathConversion* pathConversion = AZ::Interface<IPathConversion>::Get();
+
+        AZ_Assert(pathConversion, "IPathConversion interface is not available");
+
+        QString relativePath, scanfolderPath;
+        pathConversion->ConvertToRelativePath(absolutePath.FixedMaxPathStringAsPosix().c_str(), relativePath, scanfolderPath);
+        auto* scanFolderInfo = pathConversion->GetScanFolderForFile(scanfolderPath);
+
+        AZ_Assert(scanFolderInfo, "Failed to find scanfolder for " AZ_STRING_FORMAT, AZ_STRING_ARG(absolutePath.Native()));
+
+        m_scanfolderId = scanFolderInfo->ScanFolderID();
+        m_absolutePath = absolutePath;
+        m_relativePath = relativePath.toUtf8().constData();
+        m_scanfolderPath = scanfolderPath.toUtf8().constData();
+
+        Normalize();
+    }
+
+    SourceAssetReference::SourceAssetReference(AZ::IO::PathView scanFolderPath, AZ::IO::PathView pathRelativeToScanFolder)
+    {
+        AZ_Assert(!scanFolderPath.Native().empty(), "scanfolderPath is empty");
+        AZ_Assert(!pathRelativeToScanFolder.Native().empty(), "pathRelativeToScanFolder is empty");
```

Comment:
I'm on the fence about this.  If it were an error or anything else instead, it would likely happen silently and we'd never know about it or fix it, because unfortunately that's what happens in a lot of cases in AP since most things get routed to logs that no one looks at.  I like it because it helps catch bugs quickly (it already has pointed out a few) but if someone is strongly against it I can take them out.

Reply:
I've removed these since Joe also brought it up